src/components/dl-displayer-controller.tsx

```diff
@@ -34,12 +36,12 @@ export const DLDisplayerController = (): React.JSX.Element => {
 			crossAlignment="flex-start"
 			data-testid="displayer"
 		>
-			{(activeItem && (
+			{(id && (
 				<DistributionListDisplayer
 					members={members}
 					totalMembers={totalMembers}
 					distributionList={distributionList}
-					showMembersList={distributionList?.canRequireMembers ?? false}
+					key={id}
```

Comment:
Why is a key needed here?

Reply:
The key is used to make react reset the states while rerendering, based on the id value. In other words, if the id is different, the displayer knows to reset its states (from https://react.dev/learn/preserving-and-resetting-state#option-2-resetting-state-with-a-key)

Reply:
I get the point of the key but in this particular case what states needed a reset?

Reply:
Yes, the selected tab is the one I wanted to reset. By resetting the tab, I'm also avoiding an undesirable behavior inside the list of members.
Since the data about the members are in a hook which is not involved in this key-reset, if the tab was kept on the previous selection, the member list specifically, the list of members showed the previous members, until the new data was returned by the network response. Even the details are not updated immediately, but with the caching system we are going to introduce this won't be true anymore, because all the data we show in the details are already loaded from the list.
Plus, in the only other module where we have tabs inside the displayer, Files, the desiderata was to reset the tab between different nodes.
These were the considerations that make me set the key to reset the state inside the displayer.

Reply:
Thanks for the clear explanation, it makes sense now also to me.

Reply:
Can we cover this case with a test to ensure it will be kept until it reflect the wanted behaviour? Your explanation is very clear to me today but any dev could work on this file in the future and miss this usage removing the key thinking it became redundant.

Reply:
Test added https://github.com/zextras/carbonio-contacts-ui/pull/191/commits/43c2f4296d05c757f58b71ea9a6acf67ed0036b2 :heavy_check_mark: